version: '3'

services:
  planka:
    image: ghcr.io/plankanban/planka:latest
    restart: on-failure
    volumes:
      - user-avatars:/app/public/user-avatars
      - project-background-images:/app/public/project-background-images
      - attachments:/app/private/attachments
    ports:
      - 3007:1337
    environment:
      - BASE_URL=https://projects.sotcpu.me
      - DATABASE_URL=postgresql://postgres@postgres/planka
      - SECRET_KEY=
      - DEFAULT_ADMIN_EMAIL=
      - DEFAULT_ADMIN_PASSWORD=
      - DEFAULT_ADMIN_NAME=
      - DEFAULT_ADMIN_USERNAME=

  postgres:
    image: postgres:16-alpine
    restart: on-failure
    volumes:
      - db-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=
      - POSTGRES_HOST_AUTH_METHOD=trust
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres -d planka"]
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  user-avatars:
  project-background-images:
  attachments:
  db-data:

